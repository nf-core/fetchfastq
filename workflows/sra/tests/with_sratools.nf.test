nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "sra_force_sratools_download"
    tag "workflow"

    test("Force SRA-Tools Download") {

        when {
            params {
                outdir                  = "$outputDir"
                force_sratools_download = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(NftestUtils.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/fastq/").list()).match("fastq") },
                { assert snapshot(path("$outputDir/metadata/DRR026872.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/DRR028935.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/ERR1109373.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/ERR1160846.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/GSE214215.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR12848126.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR13191702.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR14593545.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR14709033.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR9984183.runinfo_ftp.tsv")).match("metadata") },
                { assert new File("$outputDir/metadata/GSM4907283.runinfo_ftp.tsv").exists() },
                { assert snapshot(path("$outputDir/samplesheet/id_mappings.csv"),
                                path("$outputDir/samplesheet/multiqc_config.yml")).match("samplesheet") },
                { assert new File("$outputDir/samplesheet/samplesheet.csv").exists() }
            )
        }

    }

}
