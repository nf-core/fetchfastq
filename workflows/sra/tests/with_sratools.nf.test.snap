{
    "metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,ef90876b2cbcb3721251b4cdef8cc47b",
            "DRR028935.runinfo_ftp.tsv:md5,f79eefcea5db346f76f76f87df7f8dd2",
            "ERR1109373.runinfo_ftp.tsv:md5,d1717cb1638950b7af22ec1b976616a4",
            "ERR1160846.runinfo_ftp.tsv:md5,1b2dc1a64117de20b2164724836da345",
            "GSE214215.runinfo_ftp.tsv:md5,23a580c89ee7bf681ca3ffa26aa1837f",
            "SRR12848126.runinfo_ftp.tsv:md5,5d1e6bfd3d4077d1e66378c58e44975d",
            "SRR13191702.runinfo_ftp.tsv:md5,409e262bb41bb6103315126db4d0dd1d",
            "SRR14593545.runinfo_ftp.tsv:md5,888eaa19fd0b7fea4dc3e9800c1d26f2",
            "SRR14709033.runinfo_ftp.tsv:md5,975dd9b32bdbd78b5da3cd88a5f705c1",
            "SRR9984183.runinfo_ftp.tsv:md5,911a48308f76387b97842d35abc43c8f"
        ],
        "timestamp": "2023-07-06T14:08:40+0000"
    },
    "samplesheet": {
        "content": [
            "id_mappings.csv:md5,aa5eab8068e59a3866fd6d1fa8fe1b5a",
            "multiqc_config.yml:md5,1337fdab76cbca9b78dbf06d7344a3b8"
        ],
        "timestamp": "2023-07-06T14:08:40+0000"
    },
    "software_versions": {
        "content": [
            "{CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, CUSTOM_SRATOOLSNCBISETTINGS={sratools=2.11.0}, MULTIQC_MAPPINGS_CONFIG={python=3.9.5}, SRA_IDS_TO_RUNINFO={python=3.9.5}, SRA_MERGE_SAMPLESHEET={sed=4.7}, SRA_RUNINFO_TO_FTP={python=3.9.5}, Workflow={nf-core/fetchngs=1.10.1}}"
        ],
        "timestamp": "2023-07-06T14:08:40+0000"
    },
    "fastq": {
        "content": [
            "DRX024467_DRR026872.fastq.gz:md5,bd3f113c3b06cf4499fa0e00c634ec85",
            "DRX026011_DRR028935_1.fastq.gz:md5,0f96fb60e903c8cbd55a6977e66f7c93",
            "DRX026011_DRR028935_2.fastq.gz:md5,b97b1e0e806bcae10a7443e2f22d6ca2",
            "ERX1188904_ERR1109373_1.fastq.gz:md5,e7edae2715dfaf4c318d37959fd9ae22",
            "ERX1188904_ERR1109373_2.fastq.gz:md5,34ab048c92201a5ec4ab36d8dc1c5732",
            "ERX1234253_ERR1160846.fastq.gz:md5,22683b24b2467494e430643423f1e275",
            "SRX10940790_SRR14593545_1.fastq.gz:md5,6a4762cf0f1f9f5544d3735e2e07c5d3",
            "SRX10940790_SRR14593545_2.fastq.gz:md5,cbd30715fdef457d7aa24a12f19eedf2",
            "SRX11047067_SRR14709033.fastq.gz:md5,79140dd1cb3fd1c5e02d88e01f7c3eda",
            "SRX17709227_SRR21711856.fastq.gz:md5,eae43f1368d4568f415581394dc9726e",
            "SRX17709228_SRR21711855.fastq.gz:md5,5194f8009b74552595c4ec5b7605d231",
            "SRX6725035_SRR9984183.fastq.gz:md5,7a70e678c400366869601c8be41fe74b",
            "SRX9315476_SRR12848126_1.fastq.gz:md5,03f5a34586aef9cd61f54eb26831a665",
            "SRX9315476_SRR12848126_2.fastq.gz:md5,97701c5d884af8d96ba95d7237f3bf79",
            "SRX9504942_SRR13055517_1.fastq.gz:md5,b6d912a73466c8f27b1d8021f80ec2bb",
            "SRX9504942_SRR13055517_2.fastq.gz:md5,7c6e220694fa5d6d9124c739aa70ea92",
            "SRX9504942_SRR13055518_1.fastq.gz:md5,ed29278c21a1bb2d8907a74be0dd4bc4",
            "SRX9504942_SRR13055518_2.fastq.gz:md5,83785b728312deb81483fc1fd5b14264",
            "SRX9504942_SRR13055519_1.fastq.gz:md5,28babc59a625b5161d97392efd4d0018",
            "SRX9504942_SRR13055519_2.fastq.gz:md5,cf1eb4c61ab3522f4d473c86b2f696e1",
            "SRX9504942_SRR13055520_1.fastq.gz:md5,cc75af896ec8a7cececb37851b0c6fcd",
            "SRX9504942_SRR13055520_2.fastq.gz:md5,fe3e2bf9f738e4959243fad59e879eae",
            "SRX9626017_SRR13191702_1.fastq.gz:md5,627a90afcd9f3cfc7f5f91074f9a2df7",
            "SRX9626017_SRR13191702_2.fastq.gz:md5,3d9b2c68a8ad1547105b17de81b2d9af"
        ],
        "timestamp": "2023-07-06T14:08:40+0000"
    }
}